src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java

```diff
@@ -266,18 +303,35 @@ public final void visitToken(DetailAST blockCommentNode) {
                 result = TREE_CACHE.get().get(treeCacheKey);
             }
             else {
-                result = context.get().parser.parseJavadocAsDetailNode(blockCommentNode);
+                final JavadocDetailNodeParser.ErrorListenerOption errorListenerOption;
+                if (violateExecutionOnNonTightHtml || !doesCheckAcceptJavadocWithNonTightHtml()) {
+                    errorListenerOption = JavadocDetailNodeParser
+                            .ErrorListenerOption.NOTIFY_FOR_NON_TIGHT_HTML;
+                }
+                else {
+                    errorListenerOption = JavadocDetailNodeParser.ErrorListenerOption.DEFAULT;
+                }
+                result = context.get().parser
+                        .parseJavadocAsDetailNode(blockCommentNode, errorListenerOption);
                 TREE_CACHE.get().put(treeCacheKey, result);
             }
 
             if (result.getParseErrorMessage() == null) {
                 processTree(result.getTree());
             }
             else {
-                final ParseErrorMessage parseErrorMessage = result.getParseErrorMessage();
-                log(parseErrorMessage.getLineNumber(),
-                        parseErrorMessage.getMessageKey(),
-                        parseErrorMessage.getMessageArguments());
+                final DetailNode javadocTree = result.getTree();
+
+                if (javadocTree != null && doesCheckAcceptJavadocWithNonTightHtml()) {
```

Comment:
@Vladlis Done. Please have a look. `enum` has been removed entirely and now `ParseStatus` stores everything.